### PR TITLE
[SCB-1494] Optimizing log when starting a new microservice

### PR DESCRIPTION
**Pull request by AngLi2**
[SCB-1494] Optimizing log when starting a new microservice

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 1]
```diff
@@ -142,6 +143,23 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
...
+                String errorCode = JsonUtils.readValue(bodyBuffer.getBytes(), Map.class).get("errorCode").toString();
+                // 400012: ErrServiceNotExists
+                // 400016: ErrSchemaNotExists
+                if (errorCode.equals("400012") || errorCode.equals("400016")) {
+                  countDownLatch.countDown();
+                  return;
+                }
+              } catch (IOException | NullPointerException ignored) {
+                // Ignore IOException when read value from bodyBuffergit
+                // Ignore NullPointerException when bodyBuffer not contains errorCode
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: Does this mean these two exception will never happen? I don't think it's not a good idea to ingore them when the exception will throw.

Comment (AngLi2) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: IOException occures when JsonUtil read value failed and NullPointerException occurs when bodyBuffer not contains errorCode. I will log some warn info to deal with them.

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 2]
```diff
@@ -142,6 +143,24 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
...
+                // 400012: ErrServiceNotExists
+                // 400016: ErrSchemaNotExists
+                if (errorCode.equals("400012") || errorCode.equals("400016")) {
+                  countDownLatch.countDown();
+                  return;
+                }
+              } catch (IOException e) {
+                LOGGER.warn("read value failed from buffer {}", bodyBuffer.toString());
+              } catch (NullPointerException e) {
+                LOGGER.warn("error code not included in buffer {}", bodyBuffer.toString());
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: Check  errorCode is null is better, never throw NPE and catch

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java (hunk 1 above)
Comment (AngLi2) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: Already modified as suggested

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java (hunk 2 above)
Comment (AngLi2) [reply service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: Modified, now check errorCode before using it

**Comment by liubao68** [comment]
Please retrigger CI after #1340  is merged or some test case will randomly fail.

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 3]
```diff
@@ -142,6 +144,24 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
               return;
             }
 
+            // no need to generate warn log when schema or service not exist
+            if (HttpStatusClass.CLIENT_ERROR.equals(HttpStatusClass.valueOf(response.statusCode()))) {
+              try {
+                if (JsonUtils.readValue(bodyBuffer.getBytes(), Map.class).containsKey("errorCode")) {
+                  @SuppressWarnings("unchecked")
```
Comment (liubao68) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: @SuppressWarnings("unchecked")  should also cover the above line.   
```
JsonUtils.readValue(bodyBuffer.getBytes(), Map.class)
```
is a big operation, it's better only call it once.

File: service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java [hunk 4]
```diff
@@ -142,6 +144,24 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
...
 
+            // no need to generate warn log when schema or service not exist
+            if (HttpStatusClass.CLIENT_ERROR.equals(HttpStatusClass.valueOf(response.statusCode()))) {
+              try {
+                Map<String, String> bufferMap = JsonUtils.readValue(bodyBuffer.getBytes(), Map.class);
+                if (bufferMap.containsKey("errorCode")) {
+                  String errorCode = bufferMap.get("errorCode");
+                  // 400012: ErrServiceNotExists
+                  // 400016: ErrSchemaNotExists
+                  if (errorCode.equals("400012") || errorCode.equals("400016")) {
```
Comment (yhs0092) [inline service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java]: Better to extract constants, like `ERR_SERVICE_NOT_EXISTS`

**Pull request merged**